apache/flink
# [FLINK-2837][storm] various improvements for Storm compatibility

This pull request contains various fixes. Most prominently, the parsing logic for Storm topologies has been changed to support multiple inputs. The API has been redone slightly. Two new examples have been added.

- refactor to use Storm's topology builder

- remove FlinkTopologyBuilder

- instantiate context-based StreamExecutionEnvironment (local or remote)

- remove Flink and Storm behavior replicating classes

- modify FlinkTopology to parse Storm topology directly

- replace StormTestBase with StreamingTestBase

- let the FiniteFileSpout finish in corner cases

- expose taskId, fix off by one task id

- add print example

- FlinkTopologyBuilder changes (check if all inputs are available before processing)

- correct package typo

- support getter methods on TupleWrapper

- two input support

- add join example

- update docs

- use Flink file system access
